### PR TITLE
Configure CircleCI to build the project and run unit tests

### PR DESCRIPTION
**Pull request by sunchunqiang**
configure circleci

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+#
+# Check https://circleci.com/docs/2.0/ios-migrating-from-1-2/ for more details
+#
+version: 2
+jobs:
+
+  build:
+    working_directory: ~/code
+    docker:
+      - image: circleci/android:api-27-alpha
```
Comment (rohandubal) [inline .circleci/config.yml]: what does the alpha here indicate? Is it some alpha build?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+          command: |
+            set -x
+            export release_tag=${CIRCLE_TAG}
+            echo "release_tag:$release_tag"
+            export version=$(echo "$release_tag" | sed 's|.*v\([0-9\.]*\).*|\1|')
+            export tagdescription=$(sed -n "/## \[Release $version\]/,/## \[Release [0-9]*\.[0-9]*\.[0-9]\]/p"  CHANGELOG.md | sed '1d' | sed '$d')
+            echo "tagdescription:$tagdescription"
+            export tagname="AWS SDK for Android v$version"
+            echo "tagname:$tagname"
+            echo "token:${GH_TOKEN}"
```
Comment (rohandubal) [inline .circleci/config.yml]: I don't this we should do this.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+      - run:
+          name: generate documents
+          command: |
+            bash gradlew generateJavadocs
+            git checkout -- gradle.properties 
+
+      - run:
+          name: copy documents
+          command: |
+            git config --local user.name "${CIRCLE_PROJECT_USERNAME}"
```
Comment (rohandubal) [inline .circleci/config.yml]: what is the value for `CIRCLE_PROJECT_USERNAME`?

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+workflows:
+  version: 2
+  build_and_test:
+    jobs: 
+      - build:
+          filters:
+            branches:
+              ignore: /gh-.*/
+  #     - unittest:
```
Comment (rohandubal) [inline .circleci/config.yml]: Can we remove the commented config?

File: CircleciScripts/maven-release.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+sonatype_username=${SonatypeUsername}
+sonatype_password=${SonatypePassword}
+gpg_keyname=${GpgKeyname}
+gpg_passphrase="${GpgPassphrase}"
+echo "gpg_passphrase:$gpg_passphrase"
+
+# Perm gen space is increased for maven command
+MAVEN_OPTS="-Xms1024m -Xmx2048m -XX:PermSize=512m -XX:MaxPermSize=1024m"
+
+mvn clean deploy -X -e -Ppublishing \
```
Comment (rohandubal) [inline CircleciScripts/maven-release.sh]: where are we getting credentials for maven deploy?

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+            python3 CircleciScripts/run_unittest.py test_results
+      - run:
+          name : check unit test result 
+          command : bash CircleciScripts/check_test_result.sh test_results
+          
+      - store_artifacts:
+          path: test_results
+
+
+  release_tag:
```
Comment (rohandubal) [inline .circleci/config.yml]: When will this step be run?

File: .circleci/config.yml [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+    working_directory: ~/code
+    docker:
+      - image: circleci/android:api-27-alpha
+    environment:
+      JVM_OPTS: -Xmx3200m
+    steps:
+      - checkout
+      - run:
+          name: build the whole project
+          command: bash gradlew build -x test
```
Comment (rohandubal) [inline .circleci/config.yml]: Can we do a build with multiple android sdk versions? This is a newer version and might not catch any api compatibility issues.

File: .circleci/config.yml (hunk 1 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: Alpha Tag
Below is from CircleCI https://circleci.com/docs/2.0/language-android/#docker-images
Our Android Docker images are currently tagged with the suffix -alpha. This is to indicate the images are currently under development and might change in backwards incompatible ways from week to week

File: .circleci/config.yml (hunk 3 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: It's a CircleCI build-in variable:  The GitHub or Bitbucket username of the current project.
 In ios project, we are using AWS. Should we use the same configuration here with iOS project?

File: .circleci/config.yml (hunk 4 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: We want to disable all jobs except to build currently

File: CircleciScripts/maven-release.sh (hunk 5 above)
Comment (sunchunqiang) [reply CircleciScripts/maven-release.sh]: from project environment variable

File: .circleci/config.yml (hunk 6 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: We want to only run build job currently.

**Comment by minbi** [comment]
@sunchunqiang Please remove all unnecessary files/scripts from this pull request for a bare bones "build". We can review the scripts and other steps as they are activated.

**Comment by minbi** [comment]
No response on pull request. Code irrelevant to this pull request is being updated on the branch that is tracked. Closing.

**Pull request closed without merging**